MISRA Compliance Updates (#71)
aws/SigV4-for-AWS-IoT-embedded-sdk: MISRA.md

# MISRA Compliance

The AWS SigV4 Library files conform to the [MISRA C:2012](https://www.misra.org.uk)
guidelines, with some noted exceptions. Compliance is checked with Coverity static analysis.
The specific deviations, suppressed inline, are listed below.

Additionally, [MISRA configuration file](https://github.com/aws/SigV4-for-AWS-IoT-embedded-sdk/blob/main/tools/coverity/misra.config) contains the project wide deviations.

### Suppressed with Coverity Comments

> To find the violation references in the source files run grep on the source code

Reviewer: ```suggestion
To find the deviation references in the source files run grep on the source code
```

Author: Put this change in, is this something we want to open a set of PR's for in the other libraries or just leave it as is for now?

Reviewer: I think we should have caught this earlier. For consistency, we should stick to either violation or deviation. But, I think it is not a blocker.

with ( Assuming rule 11.4 violation; with justification in point 2 ):
```
grep 'MISRA Ref 11.4.2' . -rI
```

#### Rule 5.4
_Ref 5.4-1_

- MISRA Rule 5.4 flags the following macro's name as ambiguous from the

>         one postfixed with _LEN. While it would be possible to change this name to

Reviewer: I think this one should be:
```suggestion
        one postfixed with _LENGTH. While it would be possible to change this name to
```

>         satisify this rule. We have chosen to suppress this singular occurence for

Reviewer: ```suggestion
        satisfy this rule. We have chosen to suppress this singular occurrence for
```
